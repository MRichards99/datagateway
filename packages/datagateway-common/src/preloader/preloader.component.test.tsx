import React from 'react';
import Preloader from './preloader.component';
import { createShallow } from '@material-ui/core/test-utils';

describe('Preloader component', () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow({});
  });

  it('renders when the site is loading', () => {
    const wrapper = shallow(<Preloader loading={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('does not render when the site is not loading', () => {
    const wrapper = shallow(<Preloader loading={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
