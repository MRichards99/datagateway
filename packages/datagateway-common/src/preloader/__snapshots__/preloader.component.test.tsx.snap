// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader component does not render when the site is not loading 1`] = `
<Preloader
  classes={
    Object {
      "container": "Preloader-container-3",
      "spinner": "Preloader-spinner-1",
      "text": "Preloader-text-4",
      "wrapper": "Preloader-wrapper-2",
    }
  }
  loading={false}
/>
`;

exports[`Preloader component renders when the site is loading 1`] = `
<Preloader
  classes={
    Object {
      "container": "Preloader-container-3",
      "spinner": "Preloader-spinner-1",
      "text": "Preloader-text-4",
      "wrapper": "Preloader-wrapper-2",
    }
  }
  loading={true}
/>
`;
